src/Xrefcheck/Verify.hs

```diff
@@ -423,29 +421,31 @@ verifyReference
   -> VerifyMode
   -> IORef VerifyProgress
   -> RepoInfo
-  -> FilePath
-  -> FilePath
+  -> CanonicalPath
   -> Reference
   -> IO (VerifyResult $ WithReferenceLoc VerifyError)
 verifyReference
   config@Config{..}
   mode
   progressRef
-  (RepoInfo files dirs)
-  root
-  fileWithReference
+  repoInfo@RepoInfo{..}
+  file
   ref@Reference{..}
-    = retryVerification 0 $ do
-        let locType = locationType rLink
-        if shouldCheckLocType mode locType
-        then case locType of
-          FileLocalLoc      -> checkRef rAnchor fileWithReference
-          RelativeLoc       -> checkRef rAnchor
-                                (normalise $ takeDirectory fileWithReference
-                                  </> toString (canonizeLocalRef rLink))
-          AbsoluteLoc       -> checkRef rAnchor (root <> toString rLink)
-          ExternalLoc       -> checkExternalResource config rLink
-          OtherLoc          -> verifying pass
+    = retryVerification 0 $
+        if shouldCheckLocType mode rInfo
+        then case rInfo of
+          RIFileLocal -> checkRef rAnchor riRoot file ""
+          RIFileRelative -> do
+            let relativeToRoot = getPosixRelativeOrAbsoluteChild riRoot (takeDirectory file)
+                  </> toString rLink
```

Comment:
Well, AFAIU, this is not always relative, from the function name. But even better it would be to call it `shownPath`, like inside the `checkRef` function, it seems that it doesn't really matter that it is relative.